sniffles/ruletrafficgenerator.py

```diff
@@ -990,15 +990,30 @@ class BackgroundTraffic(TrafficStream):
     def __init__(self, rule=None, sconf=None, start_sec=-1, start_usec=0):
         super().__init__(None, sconf)
         self.proto = rule.getProto()
-        self.sport = Port(rule.getSport())
-        self.dport = Port(rule.getDport())
+        self.protocolType = rule.getProtocolType()
         self.rule = rule
         self.rand = False
         self.current_seq_a_to_b = random.randint(0, 4000000000)
         self.current_ack_a_to_b = 0
         self.current_seq_b_to_a = random.randint(0, 4000000000)
         self.current_ack_b_to_a = 0
 
+        if self.protocolType == 'http':
+            self.sport = Port('any')
+            self.dport = Port('http')
+        elif self.protocolType == 'ftp':
+            self.sport = Port('ftp')
```

Comment:
Thank you, Dr Valgenti for the feedback. I've been mixing the ports to suit the contents (where most application protocols were server -> client). But for the new commit, I've randomized the flow (either to client or to server) and sends appropriate generic contents (server->client sends response code). But for the clients -> server cases, there wasn't any generic request codes or commands for the applications protocols (except for http) so I've left out the contents to send ACK to the server.  So I've been wondering, will the authenticity of the content matter in regard to the functionality of the background traffics?